feat: Add dynamic instruction message for checkbox

baturayo opened the pull request:
**What is the problem that this PR addresses?**
There was a static instruction message for selectbox. Thanks to this PR, we now have a way to dynamically change the instruction message thanks to `instruction` argument.

example;
```python
questionary.checkbox(
   'Select toppings',
   choices=[
       "Cheese",
       "Tomato",
       "Pineapple",
   ],
   instruction="press <enter> to approve"
).ask()
```

...

**Checklist**
<!--- Put an `x` in all the boxes that apply. -->
<!-- Automated tests validate that your PR follows our guidelines. -->
<!-- If it doesn't pass, click on it to see details as to what your PR might be missing. -->

- [x] I have read the [Contributor's Guide](https://questionary.readthedocs.io/en/stable/pages/contributors.html#steps-for-submitting-code).
- [x] I will check that all automated PR checks pass before the PR gets reviewed.

kiancross commented on questionary/prompts/checkbox.py:
```diff
@@ -104,6 +105,7 @@ def checkbox(
 
         use_emacs_keys: Allow the user to select items from the list using
                         `Ctrl+N` (down) and `Ctrl+P` (up) keys.
+        instruction: A hint on how to navigate the menu.
```
```suggestion

        instruction: A message describing how to navigate the menu.
```

kiancross replied:
```suggestion

        instruction: A message describing how to navigate the menu.
```

kiancross commented on questionary/prompts/checkbox.py:
```diff
@@ -164,15 +166,18 @@ def get_prompt_tokens() -> List[Tuple[str, str]]:
-                    "<space> to select, "
-                    "<a> to toggle, "
-                    "<i> to invert)",
+            if instruction:
```
```suggestion
            if instruction is not None:
```

kiancross reviewed (changes requested):
Thanks a lot for the PR. A couple of minor changes are below. Could you also bring the branch up-to-date with `master` (recent change which fixes the CI).

I don't think there is a good unit test that can be written for this change, is there?

baturayo commented:
Thanks a lot @kiancross , I updated the branch and committed the changes that you proposed. I also added a unit test to at least test that `instruction` does not break the flow.

kiancross reviewed (approved):
Thanks for the PR!
